File: Java/Mahasiswa.java
/*Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543 mengerjakan evaluasi Latihan 
dalam mata kuliah Design Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

//Berikut class untuk 'Mahasiswa' yang menjadi turunan dari class 'Human'
public class Mahasiswa extends Human {
    private String NIM;
    private String prodi;
    private String fakultas;

    //Empty Constructor untuk 'Mahasiswa'
    public Mahasiswa() {
        super("", "", '-');
        NIM = "";
        prodi = "";
        fakultas = "";
    }
    
    //Parameter Constructor untuk 'Mahasiswa'
    public Mahasiswa(String NIK, String nama, char jenis_kelamin, String NIM, String prodi, String fakultas) {
        super(NIK, nama, jenis_kelamin);
        this.NIM = NIM;
        this.prodi = prodi;
        this.fakultas = fakultas;
    }

    //Setter dan Getter dari Mahasiswa
    public void set_NIM(String NIM) {
        this.NIM = NIM;
    }

    public void set_prodi(String prodi) {
        this.prodi = prodi;
    }

    public void set_fakultas(String fakultas) {
        this.fakultas = fakultas;
    }

    public String get_NIM() {
        return NIM;
    }
    
    public String get_prodi() {
        return prodi;
    }

    public String get_fakultas() {
        return fakultas;
    }
}

File: C++/Mahasiswa.cpp
/*Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543 mengerjakan evaluasi Latihan 2
dalam mata kuliah Design Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

//Include untuk penggunaan string dan class Human
#include <iostream>
#include <string>
#include "Human.cpp"

//Deklarasi std agar tidak perlu menulisnya kembali pada code
using namespace std;

//Membuat class Mahasiswa dengan parent Human
class Mahasiswa : public Human
{
protected:

    //Membuat atribut yang dibutuhkan/ditambahkan dari parent
    string NIM;
    string prodi;
    string fakultas;

public:
    
    //Fungsi construct kosong
    Mahasiswa(){
        this->NIK = "";
        this->nama = "";
        this->jenis_kelmain = '-';
        this->NIM = "";
        this->prodi = "";
        this->fakultas = "";
    }

    //Fungsi construct dengan parameter
    Mahasiswa(string NIK, string nama, char jenis_kelamin, string NIM, string prodi, string fakultas){
        this->NIK = NIK;
        this->nama = nama;
        this->jenis_kelmain =  jenis_kelamin;
        this->NIM = NIM;
        this->prodi = prodi;
        this->fakultas = fakultas;
    }


    //Setter dan getter tiap atribut yang ada di class ini
    void set_NIM(string NIM)
    {
        this->NIM = NIM;
    }

    void set_prodi(string prodi){
        this->prodi = prodi;
    }

    void set_fakultas(string fakultas){
        this->fakultas = fakultas;
    }

    string get_NIM(){
        return this->NIM;
    }

    string get_prodi(){
        return this->prodi;
    }

    string get_fakultas(){
        return this->fakultas;
    }

    ~Mahasiswa(){

    }
};